modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -270,7 +270,9 @@ public Mapper parse(ParseContext context) throws IOException {
         }
 
         XContentParser parser = context.parser();
-        QueryBuilder queryBuilder = parseQueryBuilder(queryShardContext.newParseContext(parser), parser.getTokenLocation());
+        QueryBuilder queryBuilder = parseQueryBuilder(
+                queryShardContext.newParseContext(parser), parser.getTokenLocation()
```

Comment:
we still need to somehow parse the incoming queries and rewrite their source before we store them no? I mean we can do this in a different PR but we somehow need to parse Percolator queries differently if they are coming from an API or from an index. If they come from an API we need to parse with painless as default and then rewrite the source to be explicit?